calendar-tool/tool/src/java/org/sakaiproject/calendar/entityproviders/CalendarEventEntityProvider.java

```diff
@@ -112,21 +120,23 @@ public boolean entityExists(final String id) {
 		// API
 		Calendar cal;
 		try {
-			cal = calendarService.getCalendar(String.format(
-					"/calendar/calendar/%s/main", siteId));
+			//get all events from the merged calendars
+			CalendarEventVector calendarEventVector = calendarService.getEvents(calendarService.getCalendarReferences(siteId), getYearTimeRange(new CalendarUtil()));
```

Comment:
I got your point, so I should rather add some additional parameter in the url when called from Lessons and check here if the request is for merged calendars or not?